examples/tiled-rendering-example.ts

```diff
@@ -0,0 +1,394 @@
+
+/* spellchecker: disable */
+
+import { mat4, vec3, vec4 } from 'gl-matrix';
+
+import {
+    BlitPass,
+    Camera,
+    Canvas,
+    Context,
+    CuboidGeometry,
+    DefaultFramebuffer,
+    Framebuffer,
+    Invalidate,
+    MouseEventProvider,
+    Navigation,
+    Program,
+    Renderbuffer,
+    Renderer,
+    Shader,
+    Texture2D,
+    TileCameraGenerator,
+    Wizard,
+} from 'webgl-operate';
+
+import { Example } from './example';
+
+/* spellchecker: enable */
+
+
+// tslint:disable:max-classes-per-file
+
+export class TiledCubeRenderer extends Renderer {
+
+    protected _camera: Camera;
+    protected _navigation: Navigation;
+
+    protected _cuboid: CuboidGeometry;
+    protected _texture: Texture2D;
+
+    protected _program: Program;
+    protected _uViewProjection: WebGLUniformLocation;
+
+    protected _colorRenderTexture: Texture2D;
+    protected _depthRenderbuffer: Renderbuffer;
+    protected _intermediateFBO: Framebuffer;
+    protected _defaultFBO: DefaultFramebuffer;
+    protected _blit: BlitPass;
+
+    protected _tileNumber = 32;
+
+    protected _ssaaFactor = 1;
+
+    protected _tileCameraScanLineGenerator: TileCameraGenerator;
+    protected _tileCameraHilbertGenerator: TileCameraGenerator;
+    protected _tileCameraZCurveGenerator: TileCameraGenerator;
+
+    protected TIMEOUT_BETWEEN_TILES = 10; // in milliseconds
+    protected _alreadyRenderedNotTiled = false;
+
+    /**
+     * Initializes and sets up buffer, cube geometry, camera and links shaders with program.
+     * @param context - valid context to create the object for.
+     * @param identifier - meaningful name for identification of this instance.
+     * @param mouseEventProvider - required for mouse interaction
+     * @returns - whether initialization was successful
+     */
+    protected onInitialize(context: Context, callback: Invalidate,
+        mouseEventProvider: MouseEventProvider,
+        /* keyEventProvider: KeyEventProvider, */
+        /* touchEventProvider: TouchEventProvider */): boolean {
+
+        this._defaultFBO = new DefaultFramebuffer(context, 'DefaultFBO');
+        this._defaultFBO.initialize();
+        this._defaultFBO.bind();
+
+        const gl = context.gl;
+        const gl2facade = context.gl2facade;
+
+        this._colorRenderTexture = new Texture2D(this._context, 'ColorRenderTexture');
+        this._depthRenderbuffer = new Renderbuffer(this._context, 'DepthRenderbuffer');
+
+        this._intermediateFBO = new Framebuffer(this._context, 'IntermediateFBO');
+        this._blit = new BlitPass(this._context);
+        this._blit.initialize();
+        this._blit.readBuffer = gl2facade.COLOR_ATTACHMENT0;
+        this._blit.drawBuffer = gl.BACK;
+        this._blit.target = this._defaultFBO;
+        this._blit.framebuffer = this._intermediateFBO;
+
+        if (!this._intermediateFBO.initialized) {
```

Comment:
This code is within the `onInitialize` method, where the `_intermediateFBO` is set to a new instance without any condition.
I therefore suppose the FBO is always uninitialized.

Reply:
I guess you want to tell me that the if-statement is not needed. At least that would make sense to me.

Reply:
Basically, yes. The other option is that I was missing something, thus, my explicit reasoning.